src/api_manager/api_manager_impl.cc

```diff
@@ -272,6 +272,57 @@ utils::Status ApiManagerImpl::GetServiceConfigRollouts(
   return utils::Status::OK;
 }
 
+bool ApiManagerImpl::CheckBasePathMatch(const std::string &path,
+                                        const std::string &base) {
+  return (base.length() > path.length() ||
+          path.compare(0, base.length(), base) != 0 ||
+          (path.length() != base.length() && path.at(base.length()) != '/'))
+             ? false
+             : true;
+}
+
+ApiManager::ApiBasepathRewriteAction ApiManagerImpl::ReWriteURL(
+    const std::string &url, std::string *destination_url) {
+  auto server_config = global_context_->server_config();
+
+  if (server_config == nullptr ||
+      server_config->has_api_service_config() == false ||
+      server_config->api_service_config().base_path().length() == 0) {
+    return ApiBasepathRewriteAction::NONE;
+  }
+
+  std::size_t path_end_pos = url.find_first_of('?');
+  std::string path =
```

Comment:
this is exact copy.  do we need to strip out query parameters?

Reply:
Removed path variable to remove copy